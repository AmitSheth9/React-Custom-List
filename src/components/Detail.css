p {
    border: 1px solid brown;
    font-weight:700;
}
.imageD {
    width: 500px;
}
.containerD {
    border: 4px solid brown;
    width: 550px;
}
.title {
    font-size: 25px;
}