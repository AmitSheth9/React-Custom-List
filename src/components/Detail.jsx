import React from 'react'
import { move } from 'superagent'
import styles from './Detail.css'
export default function Detail({movie}) {
    return (
        <div className={styles.containerD}>
            <p className={styles.title}>{movie.title}</p>
            <img className={styles.imageD}src={`https://image.tmdb.org/t/p/w500/${movie.poster_path}`}/>
            <p>{movie.overview}</p>
            <p>Rating: {movie.vote_average}</p>
            <p>Budget: {movie.budget}</p>
            <p>Revenue: {movie.revenue}</p>
            <p>Release date: {movie.release_date}</p>
            <p>Runtime: {movie.runtime}</p>
            <p>Status: {movie.status}</p>
            <p>Tagline: {movie.tagline}</p>
        </div>
    )
}
