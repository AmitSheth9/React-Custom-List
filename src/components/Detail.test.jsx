import { render } from "@testing-library/react";
import Detail from "./Detail";

const movie = {
    title: 'Spider-Man: No Way Home',
    vote_average: 8.4,
    budget: 200000000,
    revenue: 1535856116
};

it('renders a movie', () => {
    const container = render(<Detail movie={movie} />);
    expect(container).toMatchSnapshot();
})