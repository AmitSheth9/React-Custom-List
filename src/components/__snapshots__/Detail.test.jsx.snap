// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a movie 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="containerD"
      >
        <p
          class="title"
        >
          Spider-Man: No Way Home
        </p>
        <img
          class="imageD"
          src="https://image.tmdb.org/t/p/w500/undefined"
        />
        <p />
        <p>
          Rating: 
          8.4
        </p>
        <p>
          Budget: 
          200000000
        </p>
        <p>
          Revenue: 
          1535856116
        </p>
        <p>
          Release date: 
        </p>
        <p>
          Runtime: 
        </p>
        <p>
          Status: 
        </p>
        <p>
          Tagline: 
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="containerD"
    >
      <p
        class="title"
      >
        Spider-Man: No Way Home
      </p>
      <img
        class="imageD"
        src="https://image.tmdb.org/t/p/w500/undefined"
      />
      <p />
      <p>
        Rating: 
        8.4
      </p>
      <p>
        Budget: 
        200000000
      </p>
      <p>
        Revenue: 
        1535856116
      </p>
      <p>
        Release date: 
      </p>
      <p>
        Runtime: 
      </p>
      <p>
        Status: 
      </p>
      <p>
        Tagline: 
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
